### ISSUE TITLE
Make some defaults (testrunners), overridable in the WORKSPACE

### ISSUE THREAD

**Issue by thomasvl** [bazelbuild/rules_apple#411]
Spawned out of #410

Things like the testrunners are something were the rules likely need a default, but it is very possible that a project might want all of their tests using some specific runner.  If they have to manually pass the runner to all their uses of those rules, it would be easy to miss it on a few; but them avoiding that by wrapping the rules just to override defaults also isn't great as the wrapper can then have to track changes to the rule in general.

One option might be to use an label/alias in the WORKSPACE to provide some of these defaults.  The `apple_rules_dependencies` might be able to autodetect like it does for deps or else it can take named arguments for things where the default can be swapped out.  That should allow individual WORKSPACES to swap things out.

**Comment by thomasvl** [comment]
@keith @ob fyi/thoughts for if this could work for you all?

**Comment by thomasvl** [comment]
Something like this might also work for #364 to hook in runners for apps.

**Comment by keith** [comment]
We are currently setting the `runner` field on unit and UI tests just so we can override the associated shell script to use a single simulator for multiple bundles vs creating a new one each time, which saves at least half an hour for our project. So anyway we could override this with less overhead than copying out all the shell scripts to our own project would be great.

**Comment by ob** [comment]
Yeah, this would be a good addition. I assume you could still override the test runner in the testing rule itself?

**Comment by thomasvl** [comment]
> Yeah, this would be a good addition. I assume you could still override the test runner in the testing rule itself?

Yea, it would still be an attribute you could individual set, but would just let you flip the default without having to add your own macro/rules to do so.
